usaspending_api/recipient/v2/views/list_recipients.py

```diff
@@ -70,13 +71,59 @@ def get_recipients(filters={}):
     return results, page_metadata
 
 
+def duns_count_key_function(view_instance, view_method, request, args, kwargs):
+    validated_payload = TinyShield(view_instance.models).block(request.data)
+
+    award_type = "-"
+    if "award_type" in validated_payload:
+        award_type = validated_payload["award_type"]
+
+    keyword = "-"
+    if "keyword" in validated_payload:
+        keyword = validated_payload["keyword"]
+
+    return ".".join(
+        [
+            award_type,
+            keyword,
+        ]
+    )
+
+
+class DunsCount(APIView):
+    """
+    This route takes a single keyword filter and agency_type, and returns a count of matching recipients
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/recipient/duns/count.md"
+
+    award_types = list(AWARD_TYPES.keys()) + ["all"]
+    models = [
+        {"name": "keyword", "key": "keyword", "type": "text", "text_type": "search"},
+        {"name": "award_type", "key": "award_type", "type": "enum", "enum_values": award_types, "default": "all"},
+    ]
+
+    def get_count(self, filters={}):
+        qs_filter = build_duns_base_query(filters)
+        return RecipientProfile.objects.filter(qs_filter).exclude(recipient_name__in=SPECIAL_CASES).count()
+
+    @cache_response(key_func=duns_count_key_function)
+    def post(self, request):
+        validated_payload = TinyShield(self.models).block(request.data)
+        return Response({"count": self.get_count(validated_payload)})
+
+
 class ListRecipients(APIView):
     """
     This route takes a single keyword filter (and pagination filters), and returns a list of recipients
     """
 
     endpoint_doc = "usaspending_api/api_contracts/contracts/v2/recipient/duns.md"
 
+    def request_count(self, filters={}):
+        response = DunsCount.as_view()(request=self.request._request).data
+        return response["count"]
```

Comment:
Would it be better design to handle the exceptions in this method instead of requiring a try-except wrapper below?

Reply:
See below